opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -25,40 +25,54 @@ struct ValueMap<T: Number<T>> {
     has_no_value_attribute_value: AtomicBool,
     no_attribute_value: T::AtomicTracker,
     count: AtomicUsize,
+    assign_only: bool, // if true, only assign incoming value, instead of adding to existing value
 }
 
 impl<T: Number<T>> Default for ValueMap<T> {
     fn default() -> Self {
-        ValueMap::new()
+        ValueMap::new(false)
     }
 }
 
 impl<T: Number<T>> ValueMap<T> {
-    fn new() -> Self {
+    fn new(assign_only: bool) -> Self {
         ValueMap {
             values: RwLock::new(HashMap::new()),
             has_no_value_attribute_value: AtomicBool::new(false),
             no_attribute_value: T::new_atomic_tracker(),
             count: AtomicUsize::new(0),
+            assign_only,
         }
     }
 }
 
 impl<T: Number<T>> ValueMap<T> {
     fn measure(&self, measurement: T, attrs: &[KeyValue]) {
         if attrs.is_empty() {
-            self.no_attribute_value.add(measurement);
+            if self.assign_only {
```

Comment:
@stormshield-fabs Could you help refactor this to use generics, so we dont have this duplication everywhere and get better perf?

Reply:
Let's extract this repeated snippet to a method for now? It would also make it easier to review the future PR that would refactor it using generics.